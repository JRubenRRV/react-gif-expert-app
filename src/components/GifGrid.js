import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setimages] = useState([])

    useEffect(() => { //Sirve para que se ejecute el codigo indicado solo una vez cuando se carga el componente
        getGifs();
    }, []);

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?limit=10&q=Redbullracing&api_key=9BvLBMTU3CUzTbRLcfRs0X6QC4Jbw7sk';
        const resp = await fetch(url);
        const { data } = await resp.json();

        const gifs = data.map(img => {
            return  {
                id:img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        setimages(gifs);
    }

    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
                {
                    images.map(img => 
                        <GifGridItem key={ img.id } { ...img }/>
                    )
                }
            </div>
        </>
    )
}
