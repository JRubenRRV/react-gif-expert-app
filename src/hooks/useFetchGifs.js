import { useEffect, useState } from "react"
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category)  => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => { //Sirve para que se ejecute el codigo indicado solo cuando cambia el segundo parametro
        getGifs(category)
            .then(imgs => {
                setState({
                    data: imgs,
                    loading: false
                });
            })
    }, [category]);

    return state;
}