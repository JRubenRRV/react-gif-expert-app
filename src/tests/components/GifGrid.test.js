import React from 'react';
import { shallow } from "enzyme"
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas al componente GifGrid', () => {

    const category = 'Doom';

    test('el componente debe renderizarse correctamente', () => { //Valida que se renderize el componente sin datos

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => { //Valida que se renderize el componente con datos

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/lol/what.jpg',
            title: 'fldsmdfr'
        },
        {
            id: '123',
            url: 'https://localhost/lol/what.jpg',
            title: 'fldsmdfr'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category }/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length); //Valida que el componente se pinte segun el numero de gifs indicados
        
    })
})