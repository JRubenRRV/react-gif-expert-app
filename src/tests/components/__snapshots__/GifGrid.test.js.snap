// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGrid debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Doom
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="fldsmdfr"
      url="https://localhost/lol/what.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="fldsmdfr"
      url="https://localhost/lol/what.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas al componente GifGrid el componente debe renderizarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Doom
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
